test/e2e/Helper.sol

```diff
@@ -0,0 +1,200 @@
+import {
+    GPv2Authentication,
+    GPv2Interaction,
+    GPv2Settlement,
+    GPv2Trade,
+    IERC20,
+    IVault
+} from "src/contracts/GPv2Settlement.sol";
+
+import {WETH9} from "./WETH9.sol";
+import {SettlementEncoder} from "test/libraries/encoders/SettlementEncoder.sol";
+import {SwapEncoder} from "test/libraries/encoders/SwapEncoder.sol";
+
+address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+interface IAuthorizer {
+    function grantRole(bytes32, address) external;
+}
+
+interface IERC20Mintable is IERC20 {
+    function mint(address, uint256) external;
+    function burn(uint256) external;
+}
+
+// solhint-disable func-name-mixedcase
+// solhint-disable max-states-count
+abstract contract Helper is Test {
+    using stdJson for string;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    address internal deployer;
+    address internal owner;
+    GPv2Settlement internal settlement;
+    bytes32 internal domainSeparator;
+    GPv2Authentication internal authenticator;
+    IVault internal vault;
+    GPv2AllowListAuthentication internal allowList;
+    GPv2AllowListAuthentication internal allowListImpl;
+    address vaultRelayer;
+    address balancerVaultAuthorizer;
+
+    SettlementEncoder.State internal encoder;
+    SwapEncoder.State internal swapEncoder;
+
+    address internal solver;
+    Vm.Wallet internal trader;
+
+    bool immutable isForked;
+    uint256 forkId;
+
+    WETH9 weth;
+
+    bytes32 constant SALT = "Mattresses in Berlin!";
+
+    constructor(bool _isForked) {
+        isForked = _isForked;
+    }
+
+    function setUp() public virtual {
+        if (isForked) {
+            uint256 blockNumber = vm.envUint("FORK_BLOCK_NUMBER");
+            string memory forkUrl = vm.envString("FORK_URL");
+            forkId = vm.createSelectFork(forkUrl, blockNumber);
+            weth = WETH9(payable(WETH));
+        } else {
+            weth = new WETH9();
+        }
+
+        // Configure addresses
+        deployer = makeAddr("E2E.Helper: deployer");
+        owner = makeAddr("E2E.Helper: owner");
+        solver = makeAddr("E2E.Helper: solver");
+        vm.startPrank(deployer);
+
+        // Deploy the allowlist manager
+        allowListImpl = new GPv2AllowListAuthentication{salt: SALT}();
+        allowList = GPv2AllowListAuthentication(
+            deployProxy(address(allowListImpl), owner, abi.encodeCall(allowListImpl.initializeManager, (owner)), SALT)
+        );
+        authenticator = allowList;
+
+        (balancerVaultAuthorizer, vault) = _deployBalancerVault();
+
+        // Deploy the settlement contract
+        settlement = new GPv2Settlement{salt: SALT}(authenticator, vault);
+        vaultRelayer = address(settlement.vaultRelayer());
+
+        // Reset the prank
+        vm.stopPrank();
+
+        _grantBalancerRolesToRelayer(balancerVaultAuthorizer, address(vault), vaultRelayer);
+
```

Comment:
If it's a forked test, shouldn't we use the balancer vaults that are already deployed?

Reply:
this is simpler, otherwise you'd want to use production deployment of settlement too. or make these mocks and figure out the roles on production vault.